packages/ui-kit/src/hooks/useCounter/useCounter.ts

```diff
@@ -0,0 +1,58 @@
+import { CounterQueryProps, useAccessToken, useLog } from '../../components'
+import { CounterQuery, getTimeZone, PROPEL_GRAPHQL_API_ENDPOINT, useCounterQuery } from '../../helpers'
+
+interface useCounterProps {
+  query?: CounterQueryProps
+  timeZone?: string
+}
```

Comment:
I would also suggest abstracting accessToken, metric, timeRange, filters, refetchInterval, retry, propelApiUrl query props into a common type and extending the components' props from it. For example:

Add the common type to [packages/ui-kit/src/components/shared.types.ts](https://github.com/propeldata/ui-kit/pull/132/files#diff-53d55d557d1af05e84746ea9c22bcf40fa3241326f42b3795d26ac5917aea239)

```
export type QueryProps = {
  // common props here
}
```

Then use it like
```
import { QueryProps } from 'path_to_shared.types.ts'

export interface TimeSeriesQueryProps extends QueryProps {
  /** Granularity that the chart will respond to */
  granularity?: TimeSeriesGranularity

  // other component-specific props here
}
```

Reply:
I also agree with @markandrus on reusing component props instead of introducing duplicate type definitions. In case there is a need to use a reduced amount of props, we may use [Pick](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys), e.g.
```
import { CounterQueryProps } from 'path_to_ CounterQuery_types'

export type UseCounterProps = Pick<CounterQueryProps, 'timeZone' | 'timeRange' | 'accessToken'>
```

And then expose `UseCounterProps` as API for the hook.